File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/AnalyticsQueueItem.java
package com.armongate.mobilepasssdk.model;

import com.armongate.mobilepasssdk.model.request.RequestAnalyticsData;
import com.google.gson.annotations.SerializedName;

public class AnalyticsQueueItem {
    @SerializedName("request")
    public RequestAnalyticsData request;
    
    @SerializedName("timestamp")
    public long timestamp;

    public AnalyticsQueueItem(RequestAnalyticsData request) {
        this.request = request;
        this.timestamp = System.currentTimeMillis();
    }
}

File: ios/MobilePassSDK/Model/AnalyticsQueueItem.swift
//
//  AnalyticsQueueItem.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 4.03.2025.
//

struct AnalyticsQueueItem: Codable {
    var request: RequestAnalyticsData
    var timestamp: Date
    
    init(request: RequestAnalyticsData) {
        self.request = request
        self.timestamp = Date()
    }
}
